src/socket-proxy/socket-proxyd.c

```diff
@@ -75,7 +76,30 @@ static void connection_free(Connection *c) {
 
         sd_resolve_query_unref(c->resolve_query);
 
-        free(c);
+        return mfree(c);
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(Connection*, connection_free);
+
+DEFINE_PRIVATE_HASH_OPS_WITH_VALUE_DESTRUCTOR(
+                connection_hash_ops,
+                void, trivial_hash_func, trivial_compare_func,
+                Connection, connection_free);
+
+DEFINE_PRIVATE_HASH_OPS_WITH_VALUE_DESTRUCTOR(
+                event_hash_ops,
```

Comment:
event_source_hash_ops? note that Set doesn't care if it's key/value destructor.

Reply:
Only the reason WITH_VALUE_DESTRUCTOR is used here is just I want to drop cast.